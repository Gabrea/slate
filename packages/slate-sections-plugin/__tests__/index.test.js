const path = require('path');
const compiler = require('./helpers/compiler');

jest.unmock('fs-extra');

test('sections with no seperate schemas, with liquid files that just need to be copied over', async () => {
  const stats = await compiler('fixtures/startersections/');
  expect(stats.compilation.assets).toMatchSnapshot();

  const outputFiles = stats.compilation.compiler.outputFileSystem.readdirSync(
    path.resolve(__dirname, 'dist', 'sections'),
  );
  outputFiles.forEach((file) => {
    expect(
      stats.compilation.compiler.outputFileSystem.readFileSync(
        path.resolve(__dirname, 'dist', 'sections', file),
        'utf-8',
      ),
    ).toMatchSnapshot();
  });
});

test('section that has template living in folders with schema.json and no locales', async () => {
  const stats = await compiler('fixtures/seperatejsonsections/');
  expect(stats.compilation.assets).toMatchSnapshot();

  const outputFiles = stats.compilation.compiler.outputFileSystem.readdirSync(
    path.resolve(__dirname, 'dist', 'sections'),
  );
  outputFiles.forEach((file) => {
    expect(
      stats.compilation.compiler.outputFileSystem.readFileSync(
        path.resolve(__dirname, 'dist', 'sections', file),
        'utf-8',
      ),
    ).toMatchSnapshot();
  });
});

test('sections that have templates living in folders with a schema.json and locales to go with aswell', async () => {
  const stats = await compiler('fixtures/normalsections/');
  // Check if file has been added to assets so webpack can output it
  expect(
    stats.compilation.assets['../sections/test-section.liquid'].children[0]
      ._value,
  ).toMatchSnapshot();

  // Check if file gets written to the output file system at the correct location, and with the correct values

  expect(
    JSON.parse(
      stats.compilation.assets[
        '../sections/test-section.liquid'
      ].children[1]._value
        .replace('{% schema %}', '')
        .replace('{% endschema %}', ''),
    ),
  ).toMatchSnapshot();
});
