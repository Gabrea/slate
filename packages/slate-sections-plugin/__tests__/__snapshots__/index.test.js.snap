// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`section that has template living in folders with schema.json and no locales 1`] = `
Object {
  "../sections/test-section.liquid": ConcatSource {
    "children": Array [
      RawSource {
        "_value": "
<section>
    <h2>{{ section.settings.title | escape }}</h2>
</section>
",
      },
      RawSource {
        "_value": "{% schema %}
{
  \\"name\\": \\"Blog posts\\",
  \\"settings\\": [
    {
      \\"type\\": \\"text\\",
      \\"id\\": \\"title\\",
      \\"label\\": \\"Heading\\",
      \\"default\\": \\"Blog posts\\"
    }
  ]
}
{% endschema %}",
      },
    ],
    "emitted": true,
    "existsAt": "/Users/harshalbrahmbhatt/src/github.com/Shopify/slate/packages/slate-sections-plugin/__tests__/dist/sections/test-section.liquid",
  },
  "main.js": RawSource {
    "_value": "!function(e){var t={};function r(n){if(t[n])return t[n].exports;var o=t[n]={i:n,l:!1,exports:{}};return e[n].call(o.exports,o,o.exports,r),o.l=!0,o.exports}r.m=e,r.c=t,r.d=function(e,t,n){r.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},r.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},r.t=function(e,t){if(1&t&&(e=r(e)),8&t)return e;if(4&t&&\\"object\\"==typeof e&&e&&e.__esModule)return e;var n=Object.create(null);if(r.r(n),Object.defineProperty(n,\\"default\\",{enumerable:!0,value:e}),2&t&&\\"string\\"!=typeof e)for(var o in e)r.d(n,o,function(t){return e[t]}.bind(null,o));return n},r.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(t,\\"a\\",t),t},r.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},r.p=\\"\\",r(r.s=0)}([function(e,t){}]);",
    "emitted": true,
    "existsAt": "/Users/harshalbrahmbhatt/src/github.com/Shopify/slate/packages/slate-sections-plugin/__tests__/dist/assets/main.js",
  },
}
`;

exports[`section that has template living in folders with schema.json and no locales 2`] = `
"
<section>
    <h2>{{ section.settings.title | escape }}</h2>
</section>
{% schema %}
{
  \\"name\\": \\"Blog posts\\",
  \\"settings\\": [
    {
      \\"type\\": \\"text\\",
      \\"id\\": \\"title\\",
      \\"label\\": \\"Heading\\",
      \\"default\\": \\"Blog posts\\"
    }
  ]
}
{% endschema %}"
`;

exports[`sections that have templates living in folders with a schema.json and locales to go with aswell 1`] = `
"<div class=\\"article\\" data-section-id=\\"{{ section.id }}\\" data-section-type=\\"article-template\\">
  <h1 class=\\"article__title h2\\">{{ article.title }}</h1>
</div>
"
`;

exports[`sections that have templates living in folders with a schema.json and locales to go with aswell 2`] = `
Object {
  "name": Object {
    "de": "Posts",
    "en": "Posts",
    "es": "Publicaciones",
    "fr": "Articles",
    "it": "Articoli",
    "ja": "投稿",
    "pt-BR": "posts",
  },
  "settings": Array [
    Object {
      "id": "image_parallax",
      "info": Object {
        "de": "Wird nur auf dem Desktop angezeigt",
        "en": "Only shows on desktop",
        "es": "Solo se muestra en el escritorio",
        "fr": "Ne s'affiche que sur le bureau",
        "it": "Appare solo su desktop",
        "ja": "デスクトップにのみ表示する",
        "pt-BR": "Aparece apenas na área de trabalho",
      },
      "label": Object {
        "de": "Parallax-Animation für Bild anzeigen",
        "en": "Show image parallax animation",
        "es": "Mostrar animación de paralaje de imagen",
        "fr": "Afficher l'animation en parallaxe",
        "it": "Mostra animazione parallasse immagine",
        "ja": "画像のパララックスアニメーションを表示する",
        "pt-BR": "Exibir animação de paralaxe da imagem",
      },
      "type": "checkbox",
    },
  ],
}
`;

exports[`sections with no seperate schemas, with liquid files that just need to be copied over 1`] = `
Object {
  "../sections/rich-text.liquid": RawSource {
    "_value": "
<section>
    <h2>{{ section.settings.title | escape }}</h2>
</section>

{% schema %}
  {
    \\"name\\": \\"Rich text\\",
    \\"settings\\": [
      {
        \\"type\\": \\"text\\",
        \\"id\\": \\"title\\",
        \\"label\\": \\"Heading\\",
        \\"default\\": \\"Talk about your brand\\"
      },
    ],
  }
{% endschema %}
",
    "emitted": true,
    "existsAt": "/Users/harshalbrahmbhatt/src/github.com/Shopify/slate/packages/slate-sections-plugin/__tests__/dist/sections/rich-text.liquid",
  },
  "main.js": RawSource {
    "_value": "!function(e){var t={};function r(n){if(t[n])return t[n].exports;var o=t[n]={i:n,l:!1,exports:{}};return e[n].call(o.exports,o,o.exports,r),o.l=!0,o.exports}r.m=e,r.c=t,r.d=function(e,t,n){r.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},r.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},r.t=function(e,t){if(1&t&&(e=r(e)),8&t)return e;if(4&t&&\\"object\\"==typeof e&&e&&e.__esModule)return e;var n=Object.create(null);if(r.r(n),Object.defineProperty(n,\\"default\\",{enumerable:!0,value:e}),2&t&&\\"string\\"!=typeof e)for(var o in e)r.d(n,o,function(t){return e[t]}.bind(null,o));return n},r.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(t,\\"a\\",t),t},r.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},r.p=\\"\\",r(r.s=0)}([function(e,t){}]);",
    "emitted": true,
    "existsAt": "/Users/harshalbrahmbhatt/src/github.com/Shopify/slate/packages/slate-sections-plugin/__tests__/dist/assets/main.js",
  },
}
`;

exports[`sections with no seperate schemas, with liquid files that just need to be copied over 2`] = `
"
<section>
    <h2>{{ section.settings.title | escape }}</h2>
</section>

{% schema %}
  {
    \\"name\\": \\"Rich text\\",
    \\"settings\\": [
      {
        \\"type\\": \\"text\\",
        \\"id\\": \\"title\\",
        \\"label\\": \\"Heading\\",
        \\"default\\": \\"Talk about your brand\\"
      },
    ],
  }
{% endschema %}
"
`;
